hs/src/Reach/JSUtil.hs

```diff
@@ -231,6 +265,46 @@ instance HasJSAnnot JSBinOp where
     JSBinOpTimes a -> a
     JSBinOpUrsh a -> a
 
+instance RepJSAnnot JSBinOp where
+  rjsa a' = \case
+    JSBinOpAnd _ -> JSBinOpAnd a'
+    JSBinOpBitAnd _ -> JSBinOpBitAnd a'
+    JSBinOpBitOr _ -> JSBinOpBitOr a'
+    JSBinOpBitXor _ -> JSBinOpBitXor a'
+    JSBinOpDivide _ -> JSBinOpDivide a'
+    JSBinOpEq _ -> JSBinOpEq a'
+    JSBinOpGe _ -> JSBinOpGe a'
+    JSBinOpGt _ -> JSBinOpGt a'
+    JSBinOpIn _ -> JSBinOpIn a'
+    JSBinOpInstanceOf _ -> JSBinOpInstanceOf a'
+    JSBinOpLe _ -> JSBinOpLe a'
+    JSBinOpLsh _ -> JSBinOpLsh a'
+    JSBinOpLt _ -> JSBinOpLt a'
+    JSBinOpMinus _ -> JSBinOpMinus a'
+    JSBinOpMod _ -> JSBinOpMod a'
+    JSBinOpNeq _ -> JSBinOpNeq a'
+    JSBinOpOf _ -> JSBinOpOf a'
+    JSBinOpOr _ -> JSBinOpOr a'
+    JSBinOpPlus _ -> JSBinOpPlus a'
+    JSBinOpRsh _ -> JSBinOpRsh a'
+    JSBinOpStrictEq _ -> JSBinOpStrictEq a'
+    JSBinOpStrictNeq _ -> JSBinOpStrictNeq a'
+    JSBinOpTimes _ -> JSBinOpTimes a'
+    JSBinOpUrsh _ -> JSBinOpUrsh a'
+
+instance HasJSAnnot JSArrayElement where
+  jsa = \case
+    JSArrayElement e -> jsa e
+    JSArrayComma a -> a
+
+instance RepJSAnnot JSArrayElement where
+  rjsa a' = \case
+    JSArrayElement e -> JSArrayElement $ rjsa a' e
+    JSArrayComma _ -> JSArrayComma a'
+
+instance RepJSAnnot JSObjectPropertyList where
+  rjsa a' x = (mkCommaTrailingList (map (rjsa a') $ jso_flatten x))
```

Comment:
I don't think you should be flattening and making... you should just traverse it here